internal/handlers/pg/pjson/pjson.go

```diff
@@ -229,8 +316,48 @@ func Unmarshal(data []byte) (any, error) {
 	return fromPJSON(res), nil
 }
 
-// Marshal encodes given built-in or types' package value into pjson.
-func Marshal(v any) ([]byte, error) {
+// Marshal encodes the given document and set its schema in the field $s.
+// Use it when you need to encode a document with schema, for example, when you want to store it in a database.
+func Marshal(d *types.Document) ([]byte, error) {
+	if d == nil {
+		panic("v is nil")
+	}
+
+	schema, err := makeSchema(d)
+	if err != nil {
+		return nil, err
+	}
+
+	var buf bytes.Buffer
+
+	buf.WriteString(`{"$s":`)
+	buf.Write(schema)
+
+	keys := d.Keys()
+	values := d.Values()
+
+	for i, key := range keys {
+		buf.WriteByte(',')
+		buf.WriteString(`"`)
+		buf.WriteString(key)
+		buf.WriteString(`":`)
+
+		b, err := toPJSON(values[i]).MarshalJSON()
+		if err != nil {
+			return nil, lazyerrors.Error(err)
+		}
+
+		buf.Write(b)
+	}
+
+	buf.WriteByte('}')
+
+	return buf.Bytes(), nil
+}
+
+// MarshalElem encodes given built-in or types' package value into pjson.
+// Use it when you need to encode a single value, for example in a where clause.
+func MarshalElem(v any) ([]byte, error) {
```

Comment:
Maybe we can name it `MarshalSingleValue`? It is precisely what it does and what the docs say.

Reply:
Fixed

Reply:
Awww, that suggestion gives me naming idea :) 

- `pjson.Marshal` to create a top-level json object that includes the document and the schema
- `pjson.Unmarshal` to unmarshal that top-level json object
- `MarshalSingleValue` to marshal a single value
- `unmarshalSignleValue` to unmarshal single value with the given `elem` 
- `marshalSchemaForDoc` to marshal schema based on the given doc 
- `marshalElemForSingleValue` to marshal elem for the given single value

Reply:
fixed